webapp/channels/src/components/user_account_menu/user_account_menuButton.tsx

```diff
@@ -0,0 +1,145 @@
+    openCustomStatusModal,
+    status,
+}: Props) {
+    const statusIcon = useMemo(() => {
+        if (status && status === UserStatuses.ONLINE) {
+            return (
+                <CheckCircleIcon
+                    size='16'
+                    className='userAccountMenu_onlineMenuItem_icon'
+                />
+            );
+        }
+
+        if (status && status === UserStatuses.AWAY) {
+            return (
+                <ClockIcon
+                    size='16'
+                    className='userAccountMenu_awayMenuItem_icon'
+                />
+            );
+        }
+
+        if (status && status === UserStatuses.DND) {
+            return (
+                <MinusCircleIcon
+                    size='16'
+                    className='userAccountMenu_dndMenuItem_icon'
+                />
+            );
+        }
+
+        if (status && status === UserStatuses.OFFLINE) {
+            return (
+                <RadioboxBlankIcon
+                    size='16'
+                    className='userAccountMenu_offlineMenuItem_icon'
+                />
+            );
+        }
+
+        return (
+            <ClockIcon
+                size='16'
+                className='userAccountMenu_awayMenuItem_icon'
+            />
+        );
+    }, [status]);
+
+    return (
+        <>
+            <CustomStatusEmoji
+                showTooltip={true}
+                tooltipDirection={'bottom'}
+                emojiStyle={{marginRight: '6px'}}
+                aria-hidden={true}
+                onClick={openCustomStatusModal}
+            />
+            <Avatar
+                size='sm'
+                url={profilePicture}
+                aria-hidden={true}
+            />
+            <div
+                className='userStatusIconWrapper'
+                aria-hidden={true}
+            >
+                {statusIcon}
+            </div>
+        </>
+    );
+}
+
+const ariaLabelsDefineMessages = defineMessages({
+    outOfOffice: {
+        id: 'userAccountMenu.menuButton.ariaLabel.ooo',
+        defaultMessage: 'Status is "Out of office". Open user\'s account menu.',
+    },
+    online: {
+        id: 'userAccountMenu.menuButton.ariaLabel.online',
+        defaultMessage: 'Status is "Online". Open user\'s account menu.',
+    },
+    away: {
+        id: 'userAccountMenu.menuButton.ariaLabel.away',
+        defaultMessage: 'Status is "Away". Open user\'s account menu.',
+    },
+    dnd: {
+        id: 'userAccountMenu.menuButton.ariaLabel.dnd',
+        defaultMessage: 'Status is "Do not disturb". Open user\'s account menu.',
+    },
+    offline: {
+        id: 'userAccountMenu.menuButton.ariaLabel.offline',
+        defaultMessage: 'Status is "Offline". Open user\'s account menu.',
+    },
+    notSet: {
+        id: 'userAccountMenu.menuButton.ariaLabel',
+        defaultMessage: 'Open user\'s account menu.',
+    },
+});
+
+export function getMenuButtonAriaLabel(status?: string) {
```

Comment:
i think this is appropriate since this menu item is a special one since it contains status menu item and other regular menu item. i checked with voice over, its much understandable that way